src/Grpc.AspNetCore.Server/Internal/DefaultServerCallContext.cs

```diff
@@ -0,0 +1,95 @@
+﻿#region Copyright notice and license
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Microsoft.AspNetCore.Http;
+
+namespace Grpc.AspNetCore.Server.Internal
+{
+    internal class DefaultServerCallContext : ServerCallContext
+    {
+        private readonly HttpContext _httpContext;
+        private Lazy<string> _peer;
+
+        internal DefaultServerCallContext(HttpContext httpContext)
+        {
+            _httpContext = httpContext;
+
+            var connection = _httpContext.Connection;
+            if (connection.RemoteIpAddress == null)
+            {
+                _peer = new Lazy<string>();
+            }
+            else
+            {
+                _peer = new Lazy<string>($"{(connection.RemoteIpAddress.AddressFamily == AddressFamily.InterNetwork ? "ipv4" : "ipv6")}:{connection.RemoteIpAddress}:{connection.RemotePort}");
+            }
+        }
+
+        protected override string MethodCore => _httpContext.Request.Path.Value;
+
+        protected override string HostCore => _httpContext.Request.Host.Value;
+
+        protected override string PeerCore => _peer.Value;
+
+        // TODO(JunTaoLuo, JamesNK): implement this
+        protected override DateTime DeadlineCore => throw new NotImplementedException();
+
+        protected override Metadata RequestHeadersCore => throw new NotImplementedException();
+
+        // TODO(JunTaoLuo, JamesNK): implement this
+        protected override CancellationToken CancellationTokenCore => throw new NotImplementedException();
+
+        // TODO(JunTaoLuo, JamesNK): implement this
+        protected override Metadata ResponseTrailersCore => throw new NotImplementedException();
+
+        protected override Status StatusCore { get; set; }
+
+        protected override WriteOptions WriteOptionsCore { get; set; }
```

Comment:
I need to hook this up with the HttpContextStreamWriter

Reply:
I've already done it

Reply:
What do you mean?

Reply:
Yeah, I wrote my own temp version of the context. What you've done is fine. I'll rebase the tests I added on top.

Reply:
What does your implementation look like? I found my way of doing it not quite ideal.

Reply:
https://github.com/JamesNK/grpc-dotnet/commit/ee097a4637812ceb695f38545bac21adee9948a0